src/components/RangeSlider/RangeSlider.tsx

```diff
@@ -226,118 +215,109 @@ export const RangeSlider = ({
   showSelectedRange = true,
   showHandleLabels = true,
 
-  motionBlur = false,
   springOnRelease = true,
 
   debounceInterval = 8,
-  axisLock = 'x',
-  onDrag = (newVal: number) => {
-    console.log(newVal); // eslint-disable-line no-console
-  },
+  onDrag,
+  onChange,
+  onDebounceChange,
+  onRelease,
+
   disabled = false,
   readonly = false,
   min,
   max,
   values,
   markers = [],
   testId,
+  dragHandleBehavior = 'followMouse',
 }: RangeSliderProps): JSX.Element | null => {
-  const { colors } = useTheme();
+  if (onDrag) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      'From FoundryUI RangerSlider: onDrag callback is deprecated. Instead, use onChange or onDebounceChange.',
+    );
+  }
+
+  const snapToValue = dragHandleBehavior === 'snapToValue';
+
   const { prefersReducedMotion } = useAccessibilityPreferences();
-  const hasHandleLabels = useRef(false);
-  const initializing = useRef(true);
-
-  const processedValues = useMemo(
-    () =>
-      values
-        ? // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-          // @ts-ignore This expression is not callable.
-          values.map((val: number | ValueProp) => {
-            if (typeof val === 'number') {
-              return { value: val, label: null };
-            }
-            if (Object.prototype.hasOwnProperty.call(val, 'label')) {
-              hasHandleLabels.current = true;
-            }
-            return val;
-          })
-        : [],
+  const isInitializing = useRef(true);
+
+  const debouncedOnChange = useRef(
+    debounce(newVal => {
+      if (onDrag) onDrag(newVal);
+      if (onDebounceChange) onDebounceChange(newVal);
+    }, debounceInterval),
+  ).current;
+
+  const debouncedOnRelease = useRef(
+    // wait an extra ms. onRelease should be called after onChange
+    debounce(newVal => onRelease && onRelease(newVal), debounceInterval + 1),
+  ).current;
```

Comment:
What's up with these refs? Like an alternative to useCallback?

Reply:
They're not just callbacks, they actually are callable objects (they have other funcs on them like debouncedOnChange.cancel() and stuff, which I just added disposing)